### PR TITLE
Add erlang.reference_from_dynamic.

### PR DESCRIPTION
**Pull request by NorbertSzydlik**
With these functions implemented you can use `selecting_recordX`, `selecting_anything` to receive references and subjects.

### CODE REVIEW

File: src/gleam/erlang/process.gleam [hunk 1]
```diff
@@ -71,6 +72,55 @@ pub fn new_subject() -> Subject(message) {
...
+///
+/// ```gleam
+/// import gleam/dynamic
+///
+/// subject_from_dynamic(dynamic.from(123))
+/// // -> Error([
+/// //      DecodeError(expected: "Tuple of 3 elements", found: "Int", path: []),
+/// //    ])
+/// ```
+pub fn subject_from_dynamic(
```
Comment (lpil) [inline src/gleam/erlang/process.gleam:95]: This function is unsound as it cannot verify that the message type is correct. It cannot be added to this library I'm afraid.

Comment (NorbertSzydlik) [reply src/gleam/erlang/process.gleam]: Sure, it sounds right. I removed it and left only reference_from_dynamic.

File: src/gleam/erlang/process.gleam [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import gleam/dynamic.{type DecodeErrors, type Dynamic}
 import gleam/erlang.{type Reference}
 import gleam/erlang/atom.{type Atom}
+import gleam/result
```
Comment (lpil) [inline src/gleam/erlang/process.gleam:4]: Remove this please 🙏

**Review by lpil — COMMENTED** [review]
Could you update the changelog also please!

**Comment by NorbertSzydlik** [comment]
I've added changelog. I don't know when it will be released, so I put "To be defined" as date.

**Review by lpil — APPROVED** [review]
Thank you

**Pull request merged**